Sources/SwiftProtobuf/TextFormatScanner.swift

```diff
@@ -817,15 +841,27 @@ internal struct TextFormatScanner {
             throw TextFormatDecodingError.malformedText
         }
         let c = p[0]
+        p += 1
         switch c {
-        case asciiZero, asciiOne, asciiLowerF, asciiUpperF, asciiLowerT, asciiUpperT:
-            switch parseIdentifier() {
-            case "0", "f", "false", "False":
+        case asciiZero:
+            return false
```

Comment:
Does this allow "0123" to become "0foo:1" to become "false" and then would "foo" get parsed as a field name?

Reply:
Fixed (and tested)